internal/rbd/group_controllerserver.go

```diff
@@ -166,15 +175,25 @@ func (cs *ControllerServer) DeleteVolumeGroupSnapshot(
 	// 1. verify that all snapshots in the request are all snapshots in the group
 	// 2. delete the group
 
+	groupSnapshotID := req.GetGroupSnapshotId()
```

Comment:
https://github.com/ceph/ceph-csi/blob/8dadf26537bb6108c9a8a83ab8dda962cf56dc70/internal/rbd/group_controllerserver.go#L213

Get call needs to have locks too ? 
to prevent delete and get calls at the same time.